apps/preview/next/pages/showcases/MegaMenu/MegaMenuNavigation.tsx

```diff
@@ -0,0 +1,641 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import {
+  SfIconShoppingCart,
+  SfIconFavorite,
+  SfIconPerson,
+  SfIconClose,
+  SfButton,
+  SfDrawer,
+  SfListItem,
+  useDisclosure,
+  useTrapFocus,
+  SfIconChevronRight,
+  useDropdown,
+  SfIconMenu,
+  SfCounter,
+  SfIconArrowBack,
+} from '@storefront-ui/react';
+import { type FocusEvent, Fragment, useRef, useState } from 'react';
+import brandLogo from '@assets/vsf_logo_white.svg';
+import brandLogoSign from '@assets/vsf_logo_sign_white.svg';
+import watch from '@assets/watch.png';
+import glasses from '@assets/glasses.png';
+import toy from '@assets/toy.png';
+
+const actionItems = [
+  {
+    icon: <SfIconShoppingCart />,
+    label: '',
+    ariaLabel: 'Cart',
+    role: 'button',
+  },
+  {
+    icon: <SfIconFavorite />,
+    label: '',
+    ariaLabel: 'Wishlist',
+    role: 'button',
+  },
+  {
+    icon: <SfIconPerson />,
+    label: 'Log in',
+    ariaLabel: 'Log in',
+    role: 'login',
+  },
+];
+
+type Node = {
+  key: string;
+  value: {
+    label: string;
+    counter: number;
+    link?: string;
+    banner?: string;
+    bannerTitle?: string;
+  };
+  children: Node[];
```

Comment:
Maybe we could make this type optional to make the example ~40 lines shorter? 😄 
```suggestion
  children?: Node[];
```

Reply:
We would have to make extra check every time we want to use children. This extra lines does not come with extra complexity while extra conditions do.

Reply:
Well, additional empty arrays are being allocated, but I'd say you're right - computation itself is more complex.
Should we care about this though? In this case difference is negligible so human readability imo takes importance.

But that being said - I'm not blocking my review with this comment, feel free to ignore it

Reply:
By using optional chaining `?.` we can indeed strip down a few lines on code. 👍🏼

Reply:
[`e21ea30`](https://github.com/vuestorefront/storefront-ui/pull/2640/commits/e21ea30e3e622b5c07cff0f858b31e5ec5c01111)